domain/status/storage.go

```diff
@@ -72,6 +73,39 @@ func DecodeStorageFilesystemStatus(s int) (StorageFilesystemStatusType, error) {
 	}
 }
 
+// FilesystemStatusTransitionValid returns the error
+// [statuserror.FilesystemStatusTransitionNotValid] if the transition from the
+// current status to the new status is not valid.
+func FilesystemStatusTransitionValid(
+	current StorageFilesystemStatusType,
+	isProvisioned bool,
+	new StatusInfo[StorageFilesystemStatusType],
+) error {
+	if current == new.Status {
+		return nil
+	}
+	validTransition := true
+	switch new.Status {
+	case StorageFilesystemStatusTypePending:
+		// If a filesystem is not yet provisioned, we allow its status
+		// to be set back to pending (when a retry is to occur).
+		validTransition = !isProvisioned
+	case StorageFilesystemStatusTypeError:
+		if new.Message == "" {
```

Comment:
Same comment as above. I am worried this will throw away signal and that is even worse then not have a message set.